### PR TITLE
Make UR Studio Integration Testing public

### PR DESCRIPTION
**Pull request by moriarty**
Moves some of the UR + Studio Integration Tests into this public repo
- see https://github.com/PickNikRobotics/moveit_studio/issues/4651

### CODE REVIEW

File: src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    # TODO(9427): Add support for taking snapshots prior to test execution.
+    gtest_action = objective_client_gtest
+
+    # This command terminates the process which runs Gazebo after the test cases have all finished
+    kill_gazebo_on_gtest_exit = RegisterEventHandler(
+        event_handler=OnProcessExit(
+            target_action=gtest_action,
+            on_exit=[ExecuteProcess(cmd=["killall", "ruby"])],
+        )
+    )
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py:183]: Are we doing CI with Gazebo, or is this a vestige from the time we did? Seeing if we can remove it.

File: src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                    gtest_name + "_" + timestamp_string,
+                ]
+            ),
+            "/monitored_planning_scene",
+        ],
+        output="both",
+        condition=condition_save_rosbag,
+    )
+
+    # TODO(9427): Add support for taking snapshots prior to test execution.
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py:174]: I think this number corresponds to Shortcut stories which we are no longer using, can probably remove

File: src/picknik_ur_studio_integration_testing/test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+find_package(moveit_studio_test_utils REQUIRED)
+find_package(ros_testing REQUIRED)
+
+# Integration test for ObjectiveServerNode running with other Agent nodes using real Behaviors
+ament_add_gtest_executable(test_basic_ur5e_objectives test_basic_ur5e_objectives.cpp)
+target_link_libraries(test_basic_ur5e_objectives moveit_studio_agent::do_objective_test_fixture)
+add_ros_test(launch/test_basic_ur5e_objectives.test.py TIMEOUT 120 ARGS "test_binary_dir:=${CMAKE_CURRENT_BINARY_DIR}")
+
+# Integration test for the 'getPlanningGroups' MoveGroup capability.
+ament_add_gtest_executable(test_get_planning_groups  test_get_planning_groups.cpp)
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/CMakeLists.txt:13]: ultra nit

```suggestion
ament_add_gtest_executable(test_get_planning_groups test_get_planning_groups.cpp)
```

---

... but also, I'm wondering if we even want to have this integration test here, as it's more "core" to MoveIt Studio. I would say we leave it for now, and I believe the goal for this service is to put it in OSS MoveIt.

File: src/picknik_ur_studio_integration_testing/test/launch/test_basic_ur5e_objectives.test.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    generate_agent_plus_drivers_launch_description,
+)
+
+
+def generate_test_description():
+    return generate_agent_plus_drivers_launch_description(
+        gtest_name="test_basic_ur5e_objectives",
+        env_vars={
+            # Load site config for simulated UR-5e
+            "STUDIO_CONFIG_PACKAGE": "picknik_001_ur5e_config",
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/launch/test_basic_ur5e_objectives.test.py:26]: Replace this with one of the config packages in this repo, e.g., `picknik_ur5e_site_config` or `picknik_ur5e_mock_hw_config`.

This right now only works because we still have the `picknik_ur5e_configs` repo in our submodule abyss

File: src/picknik_ur_studio_integration_testing/test/launch/test_get_planning_groups.test.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    generate_agent_plus_drivers_launch_description,
+)
+
+
+def generate_test_description():
+    return generate_agent_plus_drivers_launch_description(
+        gtest_name="test_get_planning_groups",
+        env_vars={
+            # Load site config for simulated UR-5e
+            "STUDIO_CONFIG_PACKAGE": "picknik_001_ur5e_config",
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/launch/test_get_planning_groups.test.py:26]: same here, replace with config from this repo

File: src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+
+def generate_test_description():
+    return generate_agent_plus_drivers_launch_description(
+        gtest_name="test_ur5e_pinch_p_stop",
+        env_vars={
+            # Load site config for simulated UR-5e.
+            # Note: we want a configuration without the tool_changer_link for this one as it will throw off the collision check.
+            "STUDIO_CONFIG_PACKAGE": "picknik_001_ur5e_config",
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:27]: and here too, replace with config from this repo

File: src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+  pose_name_parameter.description.name = "grasp_pose";
+  pose_name_parameter.description.type = BehaviorParameterDescription::TYPE_POSE;
+  pose_name_parameter.pose_value = grasp_pose;
+
+  do_objective_goal->parameter_overrides.insert(do_objective_goal->parameter_overrides.end(), { pose_name_parameter });
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal), 30.0));
+}
+
+TEST_F(ObjectiveFixture, DISABLED_TestInspectSurface)
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:119]: Does this test work if we enable it?

If not, can we add a comment for why?

File: src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  pose_name_parameter.joint_state_value.name = { "shoulder_pan_joint", "shoulder_lift_joint", "elbow_joint",
+                                                 "wrist_1_joint",      "wrist_2_joint",       "wrist_3_joint" };
+  pose_name_parameter.joint_state_value.position = { 0, -0.78, -0.51, -1.57, 1.65, 0 };
+
+  do_objective_goal->parameter_overrides = { pose_name_parameter };
+
+  EXPECT_TRUE(sendDoObjectiveSequenceGoal(std::move(do_objective_goal), 30.0));
+}
+
+// TODO: Enable once admittance control works with Ignition Gazebo, or we can override behaviors for sim configs.
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:172]: Since we're not using Gazebo, and it's not called Ignition anymore, maybe reword to e.g.

```suggestion
// TODO: Enable once admittance control works in the current simulation environment, or we can override behaviors for sim configs.
```

... although this should work now because overriding Objectives for sim configs should work.

File: src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+  pose_name_parameter.description.type = BehaviorParameterDescription::TYPE_POSE;
+  pose_name_parameter.pose_value = [] {
+    PoseStamped msg;
+    msg.header.frame_id = "world";
+    msg.pose.position = geometry_msgs::build<Point>().x(0.529).y(0.460).z(0.858);
+    msg.pose.orientation = geometry_msgs::build<Quaternion>().x(0.5).y(0.5).z(0.5).w(0.5);
+    return msg;
+  }();
+
+  BehaviorParameter hing_axis_pose_start_parameter;
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:193]: typo: should be `hinge_axis_pose_start_parameter`

**Review by sea-bass — CHANGES_REQUESTED** [review]
Looking great! Two major comments:

1. Change all the "proprietary and confidential" license statements at the top of files with the one we have in this repo.
2. All the integration tests are right now using the `picknik_001_ur5e_config` package which is not available in this repo. This works because this package is still in the core MoveIt Studio Docker image... for now. Consider using a package that is in this repo itself, maybe the `picknik_ur_site_config` package?

File: src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp (hunk 7 above)
Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:119]: https://github.com/PickNikRobotics/moveit_studio/pull/1111

Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:119]:
```
| 1: [test_basic_ur5e_objectives-19] [ RUN      ] ObjectiveFixture.TestInspectSurface
| 1: [test_basic_ur5e_objectives-19] 1703219439.154323 [49] test_basic: selected interface "lo" is not multicast-capable: disabling multicast
| 1: [test_basic_ur5e_objectives-19] /home/alex/workspace/moveit_studio_ws/src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:141: Failure
| 1: [test_basic_ur5e_objectives-19] Expected equality of these values:
| 1: [test_basic_ur5e_objectives-19]   set_transform_future.wait_for(std::chrono::seconds(10))
| 1: [test_basic_ur5e_objectives-19]     Which is: 4-byte object <01-00 00-00>
| 1: [test_basic_ur5e_objectives-19]   std::future_status::ready
| 1: [test_basic_ur5e_objectives-19]     Which is: 4-byte object <00-00 00-00>
| 1: [test_basic_ur5e_objectives-19] [  FAILED  ] ObjectiveFixture.TestInspectSurface (10238 ms)

```

File: src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py (hunk 1 above)
Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/picknik_ur_studio_integration_testing/generate_agent_plus_drivers_launch_description.py]: yep

File: src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp (hunk 7 above)
Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/test_basic_ur5e_objectives.cpp:141]: I fixed all your other comments and will leave this disabled.

There are 3 more test suites which were also suppose to be ported but I didn't because they were disabled I will continue working on those or create issues to do them individually and include this then.

File: src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
```
Comment (sea-bass) [inline src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: I think you missed changing this one too.

Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: No this one can't be changed it requires that config, sorry I commented and marked the conversation as resolved

Comment (sea-bass) [reply src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: I don't see a comment? https://github.com/PickNikRobotics/moveit_studio_ws/pull/158#discussion_r1434059206

What's the reason for requiring this config? Because when we remove that from the core repo, this test won't be able to find the package.

But if there is something you plan to do in a follow-on PR to address this, that's OK and we can go ahead with this.

Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: Yes I also can't find the comment

```
            # Note: we want a configuration without the tool_changer_link for this one as it will throw off the collision check.
```

Comment (moriarty) [reply src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: ah I put it in the commit message https://github.com/PickNikRobotics/moveit_studio_ws/pull/158/commits/da3b1662e09a9917918d0dfb5ad5521882c582b5

Comment (sea-bass) [reply src/picknik_ur_studio_integration_testing/test/launch/test_ur5e_pinch_p_stop.test.py:49]: I see... hmm, well, we can deal with it in future.
